localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_request.py

```diff
@@ -52,8 +52,14 @@ def __call__(
         # TODO: create helper method to render the different URI with stageVariables and parameters
         rendered_integration_uri = integration["uri"]  # use request_data_mapping["path"]
 
+        # TODO: verify the assumptions about the method with an AWS validated test
+        # if the integration method is defined and is not ANY, we can use it for the integration
+        if not (integration_method := integration["httpMethod"]) or integration_method == "ANY":
+            # otherwise, fallback to the request's method
+            integration_method = context.invocation_request["method"]
```

Comment:
That seems more accurate. But yes some AWS validated tests will help confirm later on. Good to have the TODO.